src/System.Private.CoreLib/shared/System/Buffers/Text/Utf8Parser/Utf8Parser.Date.J.cs

```diff
@@ -0,0 +1,298 @@
+                        }
+
+                        if (sourceLength - sourceIndex > 1)
+                        {
+                            if (source[sourceIndex] == Utf8Constants.Colon)
+                                state = Utf8Constants.DateJParserState.Second;
+                            else if (source[sourceIndex] == Utf8Constants.UtcOffsetChar)
+				            {
+					            offsetChar = Utf8Constants.UtcOffsetChar;
+					            continueParsing = false;
+				            }
+                            else if (source[sourceIndex] == Utf8Constants.Plus)
+                            {
+                                offsetChar = Utf8Constants.Plus;
+                                state = Utf8Constants.DateJParserState.OffsetHours;
+                            }
+                            else if (source[sourceIndex] == Utf8Constants.Minus)
+                            {
+                                offsetChar = Utf8Constants.Minus;
+                                state = Utf8Constants.DateJParserState.OffsetHours;
+                            }
+                            else
+							{
+								continueParsing = false;
+								break;
+							}
+
+                            sourceIndex++;
+                        }
+						break;
+					case Utf8Constants.DateJParserState.Second:
+						if (!ParserHelpers.TryGetNextTwoDigits(source, sourceIndex, out sourceIndex, out second))
+                        {
+                            state = Utf8Constants.DateJParserState.Invalid;
+                            break;
+                        }
+
+						if (sourceLength - sourceIndex > 1)
+                        {
+                            if (source[sourceIndex] == Utf8Constants.Period)
+                                state = Utf8Constants.DateJParserState.Fraction;
+                            else if (source[sourceIndex] == Utf8Constants.UtcOffsetChar)
+				            {
+					            offsetChar = Utf8Constants.UtcOffsetChar;
+					            continueParsing = false;
+				            }
+                            else if (source[sourceIndex] == Utf8Constants.Plus)
+                            {
+                                offsetChar = Utf8Constants.Plus;
+                                state = Utf8Constants.DateJParserState.OffsetHours;
+                            }
+                            else if (source[sourceIndex] == Utf8Constants.Minus)
+                            {
+                                offsetChar = Utf8Constants.Minus;
+                                state = Utf8Constants.DateJParserState.OffsetHours;
+                            }
+                            else
+							{
+								continueParsing = false;
+								break;
+							}
+
+                            sourceIndex++;
+                        }
+						break;
+					case Utf8Constants.DateJParserState.Fraction:
+						while (sourceIndex < sourceLength && ParserHelpers.IsDigit(source[sourceIndex]))
```

Comment:
There is pretty much the same code in https://github.com/dotnet/coreclr/blob/5786f5b2f1b25ad0ad696f3cf248b6ca299aa356/src/System.Private.CoreLib/shared/System/Buffers/Text/Utf8Parser/Utf8Parser.TimeSpan.cs#L79, maybe share it? The only difference is truncate vs abort? On the other hand if you plan to implement 'J' for timespan too, then you could reuse that code path.

Reply:
The `TryParseTimeSpanFraction` method does not return the parsed value when it aborts. Will leave as is for now.